Fixes logically dead code (Coverity)

Crazy-P opened the pull request:
Fixes reported logically dead codes by Coverity

* image.cpp: Doesn't really need any modification. But to remove the bug
report then we have to move the MAX call away from the for loop
statement.
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488639&defectInstanceId=7496346&mergedDefectId=269418

* rasterizer_gles3.cpp: Removes unnecessary elif condition since it is
checked earlier in the function
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40487856&defectInstanceId=7496284&mergedDefectId=269698

* collada.cpp: If stamement never reached due to macro ERR_CONTINUE does
the same.
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488140&defectInstanceId=7496310&mergedDefectId=10772

* navigation_mesh.cpp: Variables should always be null - however, also
checked for the very same condition in their function call. Leaving this
for review (whether the function call is necessary or not)
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488186&defectInstanceId=7496327&mergedDefectId=269606

* path_editor_plugin.cpp: If cancel is true, then it should restore the
edited value to the original provided.
http://docs.godotengine.org/en/3.0/classes/class_editorspatialgizmo.html#class-editorspatialgizmo-commit-handle


* spatial_editor_gizmos.cpp: the very condition of i >= 3 is
predetermined in the if case right before it. Thus case 1 is always '1'
and case 2 is always '-1'

* grid_map_editor.cpp: Same as above in spatial_editor_gizmos.cpp
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40486548&defectInstanceId=7496291&mergedDefectId=10762

* voxel_light_baker.cpp: Same as above in spatial_editor_gizmos.cpp
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488339&defectInstanceId=7496307&mergedDefectId=269608

* visual_server.cpp: Same as above in spatial_editor_gizmos.cpp
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488475&defectInstanceId=7496313&mergedDefectId=10749

* visual_script_expression.cpp: char '-' is already true in the switch
case mechanism. Thus it can never reach to default case.
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40487261&defectInstanceId=7496272&mergedDefectId=269822

* particles.cpp: Case 'PARAM_MAX' is unreachable due to index checking
right before the switch execution.
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488186&defectInstanceId=7496317&mergedDefectId=269980

* shader_language.cpp: Invalid index is handled in switch default case.
`type < TYPE_FLOAT && type > TYPE_VEC4` -> `(type < TYPE_FLOAT || type > TYPE_VEC4)` 
Fixes the "always false problem" in TODO comment.
https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488833&defectInstanceId=7496314&mergedDefectId=269618

Crazy-P commented:
Still need a fix for: 

https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40487888&defectInstanceId=7496305&mergedDefectId=269888

* Probably missing an assignment to err somewhere between this [platform/android/export/export.cpp#L1684-L1761](https://github.com/godotengine/godot/blob/master/platform/android/export/export.cpp#L1684-L1761) (zip error handling?)

----------

https://scan4.coverity.com/reports.htm#v36077/p14459/fileInstanceId=40488201&defectInstanceId=7496280&mergedDefectId=269790

I don't know if it is missing the NAVIGATION LOOK  handlinghere [editor/plugins/spatial_editor_plugin.cpp#L1748-L1751,](https://github.com/godotengine/godot/blob/master/editor/plugins/spatial_editor_plugin.cpp#L1748-L1751) or if that switch case is just a quick duplicated code.

nikibobi commented:
You are wrong about marshalls.cpp and resource_format_binary.cpp
The original code `val > 0x7FFFFFFF || val < -0x80000000` checks if the value is smaller than the smallest OR larger than the largest 32-bit int. If you change to AND it will never be true cause you want a number to be negative and positive at the same time.

I think you should split your pull request cause you are making many changes on many files unrelated to each other

Crazy-P commented:
No. That is incorrect.

On 32 bit build: `-0x80000000` =  2,147,483,647
On 64 bit build: `-0x80000000` =  9,223,372,036,854,775,807 (due to indirect casting via compiler)

You have to remember that `-0x80000000` = `-1 * 0x80000000` = max positive integer.

nikibobi commented:
Yes `-0x80000000` gets promoted to ` 2,147,483,647` but what I think should be done is change it to
`val > 0x7FFFFFFF || val < -0x80000000L`. All it does is check if the number is in 64 bit integer range.

Crazy-P commented:
`val > 0x7FFFFFFF || val < -0x80000000L`if u change it into that, with an or `||`, then it is always true.

We need to figure out if it is an int32 range or in the int64 range

Crazy-P commented:
Here it is more obvious why we need to check for the int32/int64 range.

https://github.com/Crazy-P/godot/blob/b82d053981bfd4636cc69bde6c4616556ca30cdc/core/io/resource_format_binary.cpp#L1311-L1319

nikibobi commented:
`val > 0x7FFFFFFF || val < -0x80000000L` is the same as either:
- `val > 2147483647`
- `val < -2147483647`

A number cannot be smaller than a negative number AND larger than a positive number

Crazy-P commented:
Ah ye. Got you point. I forgot to check for the negative value which are smaller than `min int32` , hence why it was OR.

WIll correct my mistake here. Sorry.

Crazy-P committed:
Fixes logically dead code (Coverity)

Fixes reported logically dead codes by Coverity

* image.cpp: Doesn't really need any modification. But to remove the bug
report then we have to move the MAX call away from the for loop
statement.

* rasterizer_gles3.cpp: Removes unnecessary elif condition since it is
checked earlier in the function

* collada.cpp: If stamement never reached due to macro ERR_CONTINUE does
the same.

* navigation_mesh.cpp: Variables should always be null - however, also
checked for the very same condition in their function call. Leaving this
for review (whether the function call is necessary or not)

* path_editor_plugin.cpp: If cancel is true, then it should restore the
edited value to the original provided.
http://docs.godotengine.org/en/3.0/classes/class_editorspatialgizmo.html#class-editorspatialgizmo-commit-handle

* spatial_editor_gizmos.cpp: the very condition of i >= 3 is
predetermined in the if case right before it. Thus case 1 is always '1'
and case 2 is always '-1'

* grid_map_editor.cpp: Same as above in spatial_editor_gizmos.cpp

* voxel_light_baker.cpp: Same as above in spatial_editor_gizmos.cpp

* visual_server.cpp: Same as above in spatial_editor_gizmos.cpp

* visual_script_expression.cpp: char '-' is already true in the switch
case mechanism. Thus it can never reach to default case.

* particles.cpp: Case 'PARAM_MAX' is unreachable due to index checking
right before the switch execution.

* shader_language.cpp: Invalid index is handled in switch default case.
`type < TYPE_FLOAT && type > TYPE_VEC4` -> `(type < TYPE_FLOAT || type > TYPE_VEC4`)
Fixes the "always false problem" in TODO comment.

akien-mga commented:
Thanks, looks great!
